code/controllers/subsystems/chemistry.dm

```diff
@@ -0,0 +1,59 @@
+SUBSYSTEM_DEF(chemistry)
+	name = "Chemistry"
+	wait = 20
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_CHEMISTRY
+
+	var/list/chemical_reactions = list()
+	var/list/instant_reactions_by_reagent = list()
+	var/list/distilled_reactions_by_reagent = list()
+//	var/list/fusion_reactions_by_reagent = list() // TODO: Fusion reactions as chemical reactions
+	var/list/chemical_reagents = list()
+
+/datum/controller/subsystem/chemistry/Recover()
+	log_debug("[name] subsystem Recover().")
+	chemical_reactions = SSchemistry.chemical_reactions
+	chemical_reagents = SSchemistry.chemical_reagents
+
+/datum/controller/subsystem/chemistry/Initialize()
+	initialize_chemical_reagents()
+	initialize_chemical_reactions()
+	..()
+
+/datum/controller/subsystem/chemistry/stat_entry()
+	..("C: [chemical_reagents.len] | R: [chemical_reactions.len]")
+
+//Chemical Reactions - Initialises all /decl/chemical_reaction into a list
+// It is filtered into multiple lists within a list.
+// For example:
+// chemical_reactions_by_reagent["phoron"] is a list of all reactions relating to phoron
+// Note that entries in the list are NOT duplicated. So if a reaction pertains to
+// more than one chemical it will still only appear in only one of the sublists.
+/datum/controller/subsystem/chemistry/proc/initialize_chemical_reactions()
+	var/list/paths = decls_repository.decls_of_subtype(/decl/chemical_reaction)
+	chemical_reactions.len = paths.len // Preallocate space
```

Comment:
As discussed on Discord, preallocating list size inserts a bunch of `null` entries, since you're doing `chemical_reactions += D` later you're just ending up with a list with a billion `null`s before the reaction instances.